aten/src/ATen/native/RNN.cpp

```diff
@@ -278,10 +280,14 @@ static std::vector<QuantizedCellParamsDynamic> gather_quantized_params_dynamic(
   static at::Tensor undefined;
   std::vector<QuantizedCellParamsDynamic> result;
   TORCH_CHECK(
-      params.size() % 4 == 0,
+      params.size() % 2 == 0,
       "got an incorrect number of quantized RNN parameters");
-  for (size_t i = 0; i < params.size(); i += 4) {
-    result.emplace_back(params[i], params[i + 1], params[i + 2], params[i + 3]);
+  for (size_t i = 0; i < params.size(); i += 2) {
+    auto& packed_struct_ih = cpp_custom_type_hack::cast<PackedLinearWeight>(params[i]);
```

Comment:
Check the OSS test failure. Need to add `#ifdef USE_FBGEMM`.

Reply:
I think so. The alternative is to provide a general C/C++ reference slow implementation if USE_FBGEMM is undefined. But as you mentioned before, it would be better to force the error here and tell the users this is not correct way. The similar handler is here: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/cpu/qlinear.cpp#L224-L236